### ISSUE TITLE
Allow filtering by baseline state and suppression state

### ISSUE THREAD

**Issue by lgolding** [microsoft/sarif-vscode-extension#258]
1. Add columns to display baseline state ("new", "unchanged", "updated", "absent") and suppression state ("supressed", "not suppressed").

2. The values in the baseline state column correspond _directly_ to the SARIF `result.baselineState` property.

3. By default, the baseline state column should filter to "new", "unchanged", and "updated" ("absent" issues should be hidden by default).

4. The suppression state value needs to be _computed_ from the contents of the `result.suppressions` array, using the following logic:

```
IF there are no suppressions OR all suppressions are rejected THEN
    The suppression state is "not suppressed"
ELSE
    // There is at least one suppression that has not been rejected: either it has been accepted,
    // or it is still under review, or it has no status.
    The suppression state is "suppressed"
ENDIF 
```

5. By default, the suppression state column should filter to "not suppressed" ("suppressed" issues should be hidden by default).

6. There needs to be a filter UI on both the baseline state and the suppression state column, to allow the initially hidden "absent" and "suppressed" results to be displayed.

7. The filter logic needs to combine both columns: a result is shown only if it passes through both the baseline state filter and the suppression state filter.

@michaelcfanning FYI

**Comment by jeffersonking** [comment]
Note: This is feedback for the new webview version.

**Comment by jeffersonking** [comment]
Correction, this issue reproduces in the current version (master). Re-opening.

**Comment by JelteF** [comment]
I just noticed this too. When suppressing results in Semmle like this: https://help.semmle.com/wiki/display/SD/Suppressing+alerts

Then they are still visible in the Result List in this Visual Studio Code extension, even though suppressions contains a non-empty array: 
![image](https://user-images.githubusercontent.com/1162278/87039604-b10d3a00-c1ef-11ea-98da-8cfa19f8457f.png)

A very clear improvement would already be that they would not be shown at all in this case, although a special filter button would ofcourse be better.

**Comment by michaelcfanning** [comment]
This is an important feature. We have the same problem in the VS add-in. Our thought there is that there should be a filter to allow rendering of suppressed items (and an explicit column or other indicator of whether an item is suppressed). By default, this column or filter s/be disabled.

This is the way that the VS C# error list handles things. @lgolding @jeffersonking

**Comment by lgolding** [comment]
By the logic described in the [original issue description](https://github.com/microsoft/sarif-vscode-extension/issues/258#issue-605119318), Step 4, this result should be suppressed. Isn't this just a bug? Or is the feature described in Steps 4 through 6 just NYI?

**Comment by michaelcfanning** [comment]
First, let me apologize for not reading the spec you provided in this issue, which looks good. I honestly don't understand your question, since we would expect this suppressed item (in @JelteF's example) not to be visible by default. We'd also expect a clear mechanism to opt-in to viewing this suppressed item (which should, obviously, be clearly marked as suppressed in some way, via icon, value in a column, etc.).

I'd like to talk with you and Jefferson about a time-frame for getting this working in VS Code and the VS add-in, given the importance of the scenario. You've defined the right behavior, imo, just a matter of getting to the work.

**Comment by lgolding** [comment]
To rephrase my question. "I thought we had implemented the spec above. If we have, and that issue is (unexpectedly) displayed, is that just a bug in the implementation? Or have we in fact not fully implemented the spec yet?"

It sounds like you're saying it's the latter -- the spec isn't yet fully implemented.

**Comment by JelteF** [comment]
I'm not entirely sure if I'm understanding this discussion correctly. But let me try to rephrase my situation, to make it extra clear:

In my situation the value of the "suppressions" field has no impact at all on if an item is displayed in the Results List or not. It seems to be simply ignored completely.

I originally thought this was a bug, but then found this issue and it looked like it was tracked here. Now I'm not sure anymore.

**Comment by lgolding** [comment]
Yes, absolutely I understand. This issue does indeed track the feature you are requesting (the issue you pasted should be hidden by default, but there should be a UI to reveal it). We just need to finish the work for this feature.
